### DISCUSSION TITLE
Issue with resolvedUrl in getServersideProps?

### DISCUSSION THREAD

**Discussion by olaj** [vercel/next.js#18662] [Help]
According to this issue https://github.com/vercel/next.js/issues/16407 the resolvedUrl for getServersideProps should behave like asPath in getInitialProps to basically "show what url is displayed in the browser" as i understand it.

But it doesn't seems to work correctly.

I have a dynamic page path like this /[browse]/[...appLIst]

When i do a full refresh of the page and console log resolvedUrl i get resolvedUrl: '/platform/android' correctly. But on subsequent requests via <Link>'s my resolvedUrls becomes '/platform/android?browse=platform&appList=android'.

I have tried <Link> with both as="" and href="".

```
                    <Link
                      href="/platform/android"
                    >
                      <a>{name}</a>
                    </Link>
```

Is this a bug or is this expected behavior?

**Comment by denu5** [comment]
+1 have exactly the some behaviour. @olaj could you solve this somehow? 
 
I tried all the different next/link usages described here: https://nextjs.org/docs/routing/introduction#linking-to-dynamic-paths

**Comment by anton-johansson** [comment]
I'm also experiencing issues with this. Sure, I could always strip the query part, but sometimes there are actually valid query parameters in the URL that I do want to keep. Should this become a real issue instead of a discussion thread?

**EDIT**: I decided to use the `params` property of the context parameter. It turns out that it only contains path parameters. So I use it to remove those parameters from the `resolvedUrl` string. Kind of a nasty workaround, but it works. :)
